### PR TITLE
Support the 3D convolution ops for auto_mixed_precision

### PR DESCRIPTION
**Pull request by kaixih**
Previous TF auto_mixed_precision doesn't support 3D convolutions.

Since the cuDNN enhanced the performance of 3D convolutions since v7.6.2 (https://docs.nvidia.com/deeplearning/sdk/cudnn-release-notes/rel_762.html#rel_762), this PR added the support and related tests.

fyi @nluehr @benbarsdell

### CODE REVIEW

File: tensorflow/python/grappler/auto_mixed_precision_test.py [hunk 1]
```diff
@@ -394,6 +410,55 @@ def test_conv_bn(self):
         self.assertEqual(num_to_fp32, 1)  # After FusedBatchNormV3:0
         self.assertAllClose(output_val_ref, output_val, atol=1e-3, rtol=1e-3)
 
+  @test_util.run_deprecated_v1
+  @test_util.disable_xla('This test does not pass with XLA')
+  def test_conv3d_bn(self):
+    """Test graph with convolution followed by batch norm."""
+    with compat.forward_compatibility_horizon(2019, 6, 7):
```
Comment (reedwm) [inline tensorflow/python/grappler/auto_mixed_precision_test.py]: This compat line isn't necessary, as it's already past 2019-06-07

File: tensorflow/python/grappler/auto_mixed_precision_test.py [hunk 2]
```diff
@@ -394,6 +410,55 @@ def test_conv_bn(self):
         self.assertEqual(num_to_fp32, 1)  # After FusedBatchNormV3:0
         self.assertAllClose(output_val_ref, output_val, atol=1e-3, rtol=1e-3)
 
+  @test_util.run_deprecated_v1
+  @test_util.disable_xla('This test does not pass with XLA')
+  def test_conv3d_bn(self):
```
Comment (reedwm) [inline tensorflow/python/grappler/auto_mixed_precision_test.py]: This test will fail in the GitHub tests, as they don't yet run with cuDNN 7.6.2.

You can delete them or disable them with [`@unittest.skip`](https://docs.python.org/2/library/unittest.html#skipping-tests-and-expected-failures). Or you can wait until we start running with cuDNN >= 7.6.2.

**Review by reedwm — CHANGES_REQUESTED** [review]

**Comment by gbaned** [comment]
@houtoms Could you please check reviewer comments and keep us posted. Thanks!

**Comment by houtoms** [comment]
Sure, working on it.

**Comment by houtoms** [comment]
Thx, I've made the changes.

**Review by reedwm — DISMISSED** [review]

**Comment by reedwm** [comment]
Please fix the failing tests. Looks like you need to import `unittest`.

**Review by reedwm — APPROVED** [review]

**Pull request merged**